metadata-io/src/main/java/com/linkedin/metadata/search/utils/BrowsePathUtils.java

```diff
@@ -1,40 +1,97 @@
 package com.linkedin.metadata.search.utils;
 
-import com.linkedin.common.BrowsePaths;
 import com.linkedin.common.urn.Urn;
+import com.linkedin.common.urn.UrnUtils;
 import com.linkedin.data.schema.RecordDataSchema;
-import com.linkedin.data.template.StringArray;
+import com.linkedin.metadata.Constants;
 import com.linkedin.metadata.key.ChartKey;
 import com.linkedin.metadata.key.DashboardKey;
 import com.linkedin.metadata.key.DataFlowKey;
 import com.linkedin.metadata.key.DataJobKey;
 import com.linkedin.metadata.key.DataPlatformKey;
 import com.linkedin.metadata.key.DatasetKey;
 import com.linkedin.metadata.key.GlossaryTermKey;
+import com.linkedin.metadata.key.NotebookKey;
 import com.linkedin.metadata.models.AspectSpec;
 import com.linkedin.metadata.models.EntitySpec;
 import com.linkedin.metadata.models.registry.EntityRegistry;
 import com.linkedin.metadata.utils.EntityKeyUtils;
 import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import lombok.extern.slf4j.Slf4j;
 
 
 @Slf4j
 public class BrowsePathUtils {
-  private BrowsePathUtils() {
-    //not called
+
+  public static String getDefaultBrowsePath(
+      @Nonnull Urn urn,
+      @Nonnull EntityRegistry entityRegistry,
+      @Nonnull Character dataPlatformDelimiter) throws URISyntaxException {
+
+    switch (urn.getEntityType()) {
+      case Constants.DATASET_ENTITY_NAME:
+        DatasetKey dsKey = (DatasetKey) EntityKeyUtils.convertUrnToEntityKey(urn, getKeySchema(urn.getEntityType(), entityRegistry));
+        DataPlatformKey dpKey = (DataPlatformKey) EntityKeyUtils.convertUrnToEntityKey(
+            dsKey.getPlatform(),
+            getKeySchema(dsKey.getPlatform().getEntityType(),
+                entityRegistry));
+        String datasetNamePath = getDatasetPath(dsKey.getName(), dataPlatformDelimiter);
+        return ("/" + dsKey.getOrigin() + "/" + dpKey.getPlatformName() + datasetNamePath).toLowerCase();
+      case Constants.CHART_ENTITY_NAME:
+        ChartKey chartKey = (ChartKey) EntityKeyUtils.convertUrnToEntityKey(urn, getKeySchema(urn.getEntityType(), entityRegistry));
+        return ("/" + chartKey.getDashboardTool());
+      case Constants.DASHBOARD_ENTITY_NAME: // TODO -> Improve the quality of our browse path here.
```

Comment:
i think the answer here might lie in providing a container to dashboards, data flows and datajobs rather than improve the urn-based browse path generation

Reply:
Yeah absolutely - we need containers for these.